plugins/search-react/src/extensions.tsx

```diff
@@ -0,0 +1,228 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, {
+  ReactNode,
+  createElement,
+  isValidElement,
+  useCallback,
+} from 'react';
+
+import {
+  getComponentData,
+  useElementFilter,
+  createReactExtension,
+  useAnalytics,
+} from '@backstage/core-plugin-api';
+import {
+  Result,
+  SearchResult,
+  SearchDocument,
+} from '@backstage/plugin-search-common';
+
+import { DefaultResultListItem } from './components';
+import { List, ListProps } from '@material-ui/core';
+
+/**
+ * @internal
+ * Key for result extensions.
+ */
+const SEARCH_RESULT_LIST_ITEM_EXTENSION =
+  'search.results.list.items.extensions.v1';
+
+/**
+ * @internal
+ * Returns the first extension element found for a given result, and null otherwise.
+ * @param elements - All extension elements.
+ * @param result - The search result.
+ */
+const findSearchResultListItemExtensionElement = (
+  elements: ReactNode[],
+  result: SearchResult,
+) => {
+  for (const element of elements) {
```

Comment:
I recommend using `useElementFilter` here, it helps you filter through children and find component data,  etc. It also takes care of iterating through nested arrays and fragments as well as support for feature flags

Reply:
The elements were filtered beforehand using `useElementFilter` [here](https://github.com/backstage/backstage/pull/15855/files#diff-4c59e19e90caafd5ccb90cb3236950aa33e04d44d061b4b98aa4cbd3bda8f295R160). In this function I just wanted to get their predicate and component from data. Is there a way to get elements and data all at once?

Reply:
You can select multiple things at use in the filter function, but since you need access to the props as well that's probably just gonna make things messy. I'd say keep things as is 👍